packages/auth/src/core/auth/auth_impl.ts

```diff
@@ -429,26 +432,19 @@ export class AuthImpl implements AuthInternal, _FirebaseService {
   }
 
   async validatePassword(password: string): Promise<PasswordValidationStatus> {
-    // TODO(chazzy): Implement.
-    return Promise.reject(password);
-  }
-
-  _getPasswordPolicy(): PasswordPolicy | null {
-    if (this.tenantId === null) {
-      return this._projectPasswordPolicy;
-    } else {
-      return this._tenantPasswordPolicies[this.tenantId];
+    // Fetch the password policy if one is not cached.
+    if (!this._getPasswordPolicy()) {
+      await this._updatePasswordPolicy().catch(async error => {
+        return Promise.reject(error);
+      });
     }
-  }
 
-  async _updatePasswordPolicy(): Promise<void> {
-    const response = await _getPasswordPolicy(this);
+    // Password policy will be defined after fetching.
+    const policy: PasswordPolicyInternal = this._getPasswordPolicy()!;
 
     // Check that the policy schema version is supported by the SDK.
     // TODO: Update this logic to use a max supported policy schema version once we have multiple schema versions.
-    if (
-      response.schemaVersion !== this.EXPECTED_PASSWORD_POLICY_SCHEMA_VERSION
-    ) {
+    if (policy.schemaVersion !== this.EXPECTED_PASSWORD_POLICY_SCHEMA_VERSION) {
```

Comment:
This reminds me that we may need both upper and lower bounds of the supported schema versions?

For example, if the schema version updated from 1 to 2 then 3, and we support v2 and v3 in the SDK (v1 deprecated). We will need PASSWORD_POLICY_SCHEMA_MIN_VERSION and PASSWORD_POLICY_SCHEMA_MAX_VERSION.

Reply:
I was thinking about this too. I put this in the design doc that we may use these constants later. Do you think we should implement this now or leave it as a consideration for the future?

Reply:
Let's keep it as is for now. Supporting multiple versions may overcomplicate the implementation. Let's discuss more later.